File: app/src/main/java/com/sortedqueue/programmercreek/util/ShuffleList.java
package com.sortedqueue.programmercreek.util;

import java.util.ArrayList;
import java.util.Random;

public class ShuffleList {
	public static ArrayList<String> shuffleList(ArrayList<String> randomList) {
		int n = randomList.size();
		Random random = new Random();
		random.nextInt();
		for (int i = 0; i < n; i++) {
			int change = i + random.nextInt(n - i);
			swap(randomList, i, change);
		}
		return randomList;
	}

	private static void swap(ArrayList<String> list, int i, int change) {
		String helper = list.get(i);
		list.set(i, list.get(change));
		list.set(change, helper);
	}

}

File: app/src/main/java/com/sortedqueue/programmercreek/util/ShuffleList.kt
package com.sortedqueue.programmercreek.util

import java.util.ArrayList
import java.util.Random

object ShuffleList {

    fun shuffleList(randomList: ArrayList<String>): ArrayList<String> {
        val n = randomList.size
        val random = Random()
        random.nextInt()
        for (i in 0..n - 1) {
            val change = i + random.nextInt(n - i)
            swap(randomList, i, change)
        }
        return randomList
    }


    private fun swap(list: ArrayList<String>, i: Int, change: Int) {
        val helper = list[i]
        list[i] = list[change]
        list[change] = helper
    }

}
